framework/src/meshgenerators/MeshDiagnosticsGenerator.C

```diff
@@ -335,6 +343,66 @@ MeshDiagnosticsGenerator::checkWaterTightSidesets(const std::unique_ptr<MeshBase
   diagnosticsLog(message, _check_watertight_sidesets, num_faces_without_sideset);
 }
 
+void
+MeshDiagnosticsGenerator::checkWatertightNodesets(const std::unique_ptr<MeshBase> & mesh) const
+{
+  /*
+  Diagnostic Overview:
+  1) Mesh precheck
+  2) Loop through all elements
+  3) Loop through all sides of that element
+  4) If side is external loop through its nodes
+  5) If node is not associated with any nodeset add to list
+  6) Print out node id
+  */
+  if (mesh->mesh_dimension() < 2)
+    mooseError("The nodeset check only works for 2D and 3D meshes");
+  auto & boundary_info = mesh->get_boundary_info();
+  boundary_info.build_side_list();
+  const auto nodeset_map = boundary_info.get_nodeset_map();
+  unsigned int num_nodes_without_nodeset = 0;
+  std::set<Node> checked_nodes;
+
+  for (const auto elem : mesh->active_element_ptr_range())
+  {
+    for (const auto i : elem->side_index_range())
+    {
+      // Check if side is external
+      if (elem->neighbor_ptr(i) == nullptr)
+      {
+        // Side is external, now check nodes
+        auto side = elem->side_ptr(i);
+        const auto & node_list = side->get_nodes();
+        for (unsigned int j = 0; j < side->n_nodes(); j++)
+        {
+          const auto node = node_list[j];
+          if (checked_nodes.count(*node))
+            continue;
+          // if node is not part of nodeset map add it to list of bad nodes
+          if (nodeset_map.count(node) == 0)
+          {
+            // This node does not have a nodeset!!!
+            num_nodes_without_nodeset++;
+            checked_nodes.insert(*node);
```

Comment:
Thanks! I was trying to figure out why it was failing that test. I think I have the same issue in #28701 so I'll fix it in that one as well.

Reply:
I didnt remember that one was still open. Do you need anything else from me for that? Are we waiting on Roy's review?

Reply:
I think I'm just waiting on Roy's review. Though if this one get's merged first I might have to make some changes to the other before it can be merged.